planet/src/main/webapp/places.js

```diff
@@ -22,48 +24,60 @@ function initMap() {
         center: TORONTO_COORDINATES,
         zoom: 8
     });
-    let infoWindow = new google.maps.InfoWindow;
-        // Gets users current position
-        if (navigator.geolocation) {
-            navigator.geolocation.getCurrentPosition(function(position) {
-            let pos = {
-              lat: position.coords.latitude,
-              lng: position.coords.longitude
+
+    // Gets users current location
+    if (navigator.geolocation) {
+        navigator.geolocation.getCurrentPosition(function(position) {
+            pos = {
+            lat: position.coords.latitude,
+            lng: position.coords.longitude
             };
-            infoWindow.setPosition(pos);
-            infoWindow.setContent('You are here.');
-            infoWindow.open(map);
+            // Create draggable marker at current location
+            let searchMarker = new google.maps.Marker({
+                position: pos,
+                map: map,
+                animation: google.maps.Animation.DROP,
+                draggable: true
+            });
             map.setCenter(pos);
         }, function() {
-            // User did not allow location
-            handleLocationError(true, infoWindow, map.getCenter());
-          });
-        } else {
-            // Browser doesn't support Geolocation
-            handleLocationError(false, infoWindow, map.getCenter());
-        }
-    updateSearch();
-}
+            let searchMarker = new google.maps.Marker({
+                position: TORONTO_COORDINATES,
+                map: map,
+                animation: google.maps.Animation.DROP,
+                draggable: true
+            });
+            map.setCenter(TORONTO_COORDINATES);
+            });
+    } else {
+        alert('Error: Your browser doesn\'t support geolocation.'); 
+    }
 
-/** Create new markers and display new search */
-function updateSearch() {    
     let input = document.getElementById('pac-input');
     let searchBox = new google.maps.places.SearchBox(input);
 
     // Set position of the search bar onto the map
     map.controls[google.maps.ControlPosition.TOP_LEFT].push(input);
 
+    updateSearch();
+}
+
+/** Create new markers and display new search */
+function updateSearch() {    
```

Comment:
this function is very long. Consider breaking it up into smaller functions to improve readability.

Reply:
For example, the innermost loop which is executed for each place could be its own function